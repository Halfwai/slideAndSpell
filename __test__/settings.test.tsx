import React from 'react';
import { render } from '@testing-library/react-native';
import Settings from '@/app/settings';

it(`Settings Component renders correctly`, () => {
    const { toJSON } = render(
        <Settings/>
    );
    expect(toJSON()).toMatchSnapshot();
});