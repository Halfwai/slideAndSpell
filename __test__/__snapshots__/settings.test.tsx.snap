// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CBE6F7",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "height": "10%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Settings
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 8,
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          undefined,
        ]
      }
    >
      Change Email
    </Text>
    <TextInput
      autoComplete="email"
      autoCorrect={false}
      inputMode="email"
      onChangeText={[Function]}
      placeholder="New Email"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          undefined,
        ]
      }
    >
      Change Display Name
    </Text>
    <TextInput
      autoComplete="email"
      autoCorrect={false}
      inputMode="email"
      onChangeText={[Function]}
      placeholder="New Display Name"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          undefined,
        ]
      }
    >
      Change Password
    </Text>
    <TextInput
      autoComplete="email"
      autoCorrect={false}
      inputMode="email"
      onChangeText={[Function]}
      placeholder="New Password"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <TextInput
      autoComplete="email"
      autoCorrect={false}
      inputMode="email"
      onChangeText={[Function]}
      placeholder="Confirm Password"
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 50,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Save Changes
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#AAA",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Play Tutorial
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#AAA",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Privacy Policy
      </Text>
    </View>
  </View>
</View>
`;
